File: src/main/java/com/senderman/anitrackerbot/DBService.java
package com.senderman.anitrackerbot;

import java.util.Map;

public interface DBService {

    void saveAnime(int id, int userId, String url);

    String getAnimeUrl(int id, int userId);

    void deleteAnime(int id, int userId);

    void dropUser(int userId);

    int totalAnimes(int userId);

    boolean idExists(int id, int userId);

    boolean urlExists(String url, int userId);

    Map<Integer, String> getAllAnimes(int userId);
}

File: src/main/java/com/senderman/anitrackerbot/DBService.kt
package com.senderman.anitrackerbot

interface DBService {
    fun saveAnime(id: Int, userId: Int, url: String)
    fun getAnimeUrl(id: Int, userId: Int): String
    fun deleteAnime(id: Int, userId: Int)
    fun dropUser(userId: Int)
    fun totalAnimes(userId: Int): Int
    fun idExists(id: Int, userId: Int): Boolean
    fun urlExists(url: String, userId: Int): Boolean
    fun getAllAnimes(userId: Int): Map<Int, String>
    fun getUsersIds(): List<Int>
}